homeassistant/components/tami4/config_flow.py

```diff
@@ -82,8 +82,11 @@ async def async_step_otp(
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
             else:
+                device_name = api.device_metadata.name
+                if device_name is None:
+                    device_name = "Tami4"
```

Comment:
We require 100% test coverage of the config flow. Please add a test, where the API return no device name

Reply:
@Guy293 
Can't add a test as the DeviceMetadata doesn't support None as name

Reply:
You can turn off type-checking on that line until the typing is fixed. So you can write a test for it

Reply:
> @Guy293 Can't add a test as the DeviceMetadata doesn't support None as name

Should be a very easy fix in the api itself, you can maybe add a linter ignore comment for now...

Reply:
Thanks, Done.